### ISSUE TITLE
(1.12.2) Cascading world generation

### ISSUE THREAD

**Issue by Sutika01** [Tau-ri-Dev/Mod-JSG-1.12.2#340]
This bug is happening when I create a 1.12.2 minecraft forge server and start exploring a little bit and the server generates a stargate structure, it fills the log and causes a lot of lag that can kick you out of the server and even "kill" the server it self.

**To Reproduce**
Just make a forge server with the just stargate mod and start going any direction

**Mod version**
jsg-1.12.2-4.11.2.6-client.jar
Forge-14.23.5.2860
**Server Log**
[debug.log](https://github.com/user-attachments/files/16400877/debug.log)

**Comment by Fredyman95** [comment]
Server specifications?

**Comment by Sutika01** [comment]
In my batch file I'm using these arguments: Xmx8G Xms2G
and I'm running this on my own computer

**Comment by Sutika01** [comment]
A friend tried to run it as well, but got the same error, here are his specifications (top one CPU, bottom is the RAM), and a video of the occurance.
![image](https://github.com/user-attachments/assets/cc372b16-1778-4233-abc4-b3ca4b63a53b)
[https://www.dropbox.com/scl/fi/7lz84lgjvadq4fod88i9f/Test.mp4?rlkey=nxfhjsjv5kvesmz20kzfl332x&st=nhl1nitw&dl=0](url)

**Comment by Fredyman95** [comment]
Cascading worldgen lag - is common issue on 1.12.2. Lots of mod that add strucutres generates this message.

**Comment by Sutika01** [comment]
I agree completely, but let's say that we are making a modpack for this version, with for example Biomes o plenty, Tetra, and a couple of others in addition to JSG. In this instance, if a cascading worldgen is started by any of these mods, is the only option to avoid this happening again, is to disable half of these mods structure generating properties in the config? 

Because if one of the mods starts a cascading worldgen lag, then all of the others are going to join in and lag the server so severely, that it just dies. Even if we only moved 100-200 blocks in a direction.

We tried looking online everywhere for an answer, but either didn't get a solution, or got the suggestion to take it up with the mod's developers.
